saleor/core/views.py

```diff
@@ -32,6 +32,9 @@ def styleguide(request):
 
 
 def impersonate(request, uid):
+    if request.user.is_impersonate:
+        stop_impersonate(request)
+        request.user = request.impersonator
```

Comment:
I'm not sure about those changes. (Sorry I don't have enough time to test this right now).

From the [source code](https://bitbucket.org/petersanchez/django-impersonate/src/ba74bb2af5adbfb4219434ed7c14ffd486a71eb7/impersonate/views.py?at=default&fileviewer=file-view-default#views.py-48:79), I feel like it is not needed to call this view, it doesn't seem to be doing anything needed to replace the currently impersonated user, as what we want to do here is to replace the currently impersonated user by a new UID. And calling something that acts as a view seems really odd.